From 3a6954cff996e5f8ac4652c3b9786d25210b6afa Mon Sep 17 00:00:00 2001
From: Rafael Aquini <aquini@redhat.com>
Date: Wed, 19 Nov 2014 18:21:00 -0500
Subject: [mm] do not overwrite reserved pages counter at show_mem()

Message-id: <14f2d6b609b5db8a97ceb36438bd28be8ee08f57.1416421070.git.aquini@redhat.com>
Patchwork-id: 100374
O-Subject: [RHEL7 PATCH] mm: do not overwrite reserved pages counter at show_mem()
Bugzilla: 1125433
RH-Acked-by: Motohiro Kosaki <mkosaki@redhat.com>
RH-Acked-by: Johannes Weiner <jweiner@redhat.com>
RH-Acked-by: Larry Woodman <lwoodman@redhat.com>
RH-Acked-by: Rik van Riel <riel@redhat.com>

Minor fixlet to perform the reserved pages counter aggregation
for each node, at show_mem()

Upstream status: sent & waiting for inclusion -- https://lkml.org/lkml/2014/11/14/547

This is a follow-up fix for Bug 1125433

Signed-off-by: Rafael Aquini <aquini@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/lib/show_mem.c b/lib/show_mem.c
index 5847a49..19b6190 100644
--- a/lib/show_mem.c
+++ b/lib/show_mem.c
@@ -31,7 +31,7 @@ void show_mem(unsigned int filter)
     continue;
 
    total += zone->present_pages;
-   reserved = zone->present_pages - zone->managed_pages;
+   reserved += zone->present_pages - zone->managed_pages;
 
    if (is_highmem_idx(zoneid))
     highmem += zone->present_pages;
-- 
1.7.1